packages/runtime/container-runtime/src/containerRuntime.ts

```diff
@@ -192,10 +188,39 @@ export interface ContainerRuntimeMessage {
     type: ContainerMessageType;
 }
 
+export interface ISummaryConfigurationStartupSettings {
+    /* Delay before first attempt to spawn summarizing container. */
+    initialSummarizerDelayMs: number;
+
+    /**
+     * Flag that will enable changing elected summarizer client after maxOpsSinceLastSummary.
+     * This defaults to false (disabled) and must be explicitly set to true to enable.
+     */
+    summarizerClientElection: boolean;
+}
+export interface ISummaryConfigurationMainSettings extends ISummaryConfigurationStartupSettings {
+    idleTime: number;
+    maxTime: number;
+    maxOps: number;
+    maxAckWaitTime: number;
+    maxOpsSinceLastSummary: number;
+}
+
+export type ISummaryConfiguration =
+{
+    state: "disabled";
+} | ({
+    state: "disableHeuristics";
+    maxAckWaitTime: number;
+    maxOpsSinceLastSummary: number; } & ISummaryConfigurationStartupSettings
+    )
+  | ({ state: "enabled";} & ISummaryConfigurationMainSettings);
```

Comment:
Is this the best split of interfaces? A couple of properties like `maxAckWaitTime` and `maxOpsSinceLastSummary` are duplicated and not directly tied to a summary interface.
Would it make more sense to have the split like this:

```
export interface ISummaryConfigurationHeuristicSettings extends ISummaryConfigurationBaseSettings {
    idleTime: number;
    maxTime: number;
    maxOps: number;
}

export interface ISummaryConfigurationBaseSettings  {
    /* Delay before first attempt to spawn summarizing container. */
    initialSummarizerDelayMs: number;

    /**
     * Flag that will enable changing elected summarizer client after maxOpsSinceLastSummary.
     * This defaults to false (disabled) and must be explicitly set to true to enable.
     */
    summarizerClientElection: boolean;

    maxAckWaitTime: number;
    maxOpsSinceLastSummary: number;
}
```

Reply:
I like it :)